aya/src/programs/extension.rs

```diff
@@ -146,9 +148,21 @@ impl Extension {
     pub fn detach(&mut self, link_id: ExtensionLinkId) -> Result<(), ProgramError> {
         self.data.links.remove(link_id)
     }
+
+    /// Takes ownership of the link referenced by the provided link_id.
+    ///
+    /// The link will be detached on Drop and the caller is now responsible
```

Comment:
`Drop` or link to https://doc.rust-lang.org/stable/std/ops/trait.Drop.html whatever you prefer